src/system/chaos.291

```diff
@@ -2426,29 +2426,49 @@ BBLK
 	 JRST [	AOS CHNLOS	;No buffers, lose this packet
 		JRST CHXBK1 ]
 	HRLI A,-%CPKDT		;Copy out the packet header
+IFN KLH10P,[
+;; The KLH10P-marked code is a workaround for a bug in the 
+;; KLH10 implementation of the CH11. It seems to be a synchronization
+;; bug which results in packets being corrupted after being received
+;; from the net, and before being read by the 10.
+;; The workaround is to compute the checksum here, and verify it.
+;; When someone either fixes the CH11 bug or simply reimplements it,
+;; this code should be taken out again.
+;; Note that the performance impact is very small.
+;; /BV November 2024.
+;; (Side note: is "ADDI J,(B)" faster than "ADD J,B"?)
```

Comment:
On some machines, potentially yes.  The reason being that the immediate operand can use the possibly streamlined 18-bit addressing circuitry.  I believe it is a little faster on a KA10, or maybe a PDP-6.  Certainly this type of optimization can be seen in early MIT code.

Reply:
I don't think ITS source code comments are the right place for a Q and A, though.